pyoanda/order.py

```diff
@@ -55,9 +56,12 @@ def check(self):
                 type(self.expiry), datetime)
             raise TypeError(msg)
 
-        if hasattr(self, "price") and not isinstance(self.price, (int, float)):
-            msg = "Expiry must be int or float, '{0}' found".format(
-                type(self.price))
-            raise TypeError(msg)
+        if hasattr(self, "price"):
+            try:
+                Decimal(self.price)                
```

Comment:
I don't think it's a big performance impact, it just increases your dependencies when the float() check is probably fine.  Up to you though :)